challenges/left_join/hash_table.py

```diff
@@ -0,0 +1,77 @@
+from .linked_list import LinkedList
+
+
+class HashTable:
+    """Representation of a simple hash table."""
+    def __init__(self, max_size=1024):
+        if type(max_size) is not int:
+            raise TypeError('argument must be type <int>')
+        if max_size < 1:
+            raise ValueError('argument must be non-negative <int>')
+
+        self.max_size = max_size
+        self.buckets = [LinkedList() for _ in range(max_size)]
+
+    def __iter__(self):
+        keylist = []
+        for bucket in self.buckets:
+            cur = bucket.head
+            while cur:
+                for key in cur.val.keys():
+                    keylist.append(key)
+                cur = cur._next
+        return iter(keylist)
```

Comment:
Just so you know, you could implement this as a generator instead to avoid creating a list of all the keys in the `HashTable`.